### PR TITLE
Add update strategy and node selector overlay to Antrea 1.2.3 package

### PR DESCRIPTION
**Pull request by blc1996**
Signed-off-by: Lucheng Bao <luchengb@vmware.com>

## What this PR does / why we need it

We need to enforce the update strategy of all the deployment and daemonset running on Supervisor cluster. Also, add nodeSelector configuration to deployments.

## Which issue(s) this PR fixes

Fixes: # https://github.com/vmware-tanzu/tanzu-framework/issues/1850

## Describe testing done for PR

Manually tested the templates

## Special notes for your reviewer

I understand that Antrea intentionally set the update strategy of deployment as reCreate to avoid running 2 instances of Antrea controller pods at the same time. However, it is required by superviser cluster in vSphere that we must enforce these strategies. This is the current implementation in Supervisor cluster's guest cluster controller. Manual testing also didn't show immediate error.

### CODE REVIEW

**Review by shivaani0505 — DISMISSED** [review]
LGTM

**Review by knabben — APPROVED** [review]

File: addons/packages/antrea/1.2.3/test/antrea_test.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 	"strings"
 
 	corev1 "k8s.io/api/core/v1"
+	appsv1 "k8s.io/api/apps/v1"
```
Comment (seemiller) [inline addons/packages/antrea/1.2.3/test/antrea_test.go:11]: @blc1996 This file is failing the Golang linter check here (I believe these imports should be alphabetized). You can run the check yourself with `make lint` to see the error.

**Review by seemiller — CHANGES_REQUESTED** [review]

File: addons/packages/antrea/1.2.3/test/antrea_test.go (hunk 1 above)
Comment (blc1996) [reply addons/packages/antrea/1.2.3/test/antrea_test.go]: Fixed

**Review by seemiller — APPROVED** [review]

**Pull request merged**